### PR TITLE
Coverity

### PR DESCRIPTION
**Pull request by serval2412**
Provide a general summary of your changes in the **Title** above.

### Important
Mark with [x] to select. Leave as [ ] to unselect.

### Motivation and Context
- [ ] Why is this change required? What problem does it solve?
- [ ] If it fixes an open issue, include the text `Closes #1` (where 1 would be the issue number) to your commit message.

### Types of changes
What types of changes does your code introduce? Check all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Description
- [ ] Describe your changes in detail.

### Final checklist
Go over all the following points and check all the boxes that apply. 
If you're unsure about any of these, don't hesitate to ask. We're here to help! Various areas of the codebase have been worked on by different people in recent years, so if you are unfamiliar with the general area you're working in, please feel free to chat with people who have experience in that area. See the list [here](https://github.com/fontforge/fontforge/blob/master/CONTRIBUTING.md#people-to-ask).
- [ ] My code follows the code style of this project found [here](https://github.com/fontforge/fontforge/blob/master/CONTRIBUTING.md#coding-style).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/fontforge/fontforge/blob/master/CONTRIBUTING.md) guidelines.

Your pull request will be tested via Travis CI to automatically indicate that your changes do not prevent compilation. FontForge is a big program, so Travis can easily take over 20 minutes to confirm your changes are buildable. Please be patient. More details about using Travis can be found [here](https://github.com/fontforge/fontforge/blob/master/CONTRIBUTING.md#using-travis-ci).  
  
If it reports back that there are problems, you can log into the Travis system and check the log report for your pull request to see what the problem was. If no error is shown, just re-run the Travis test for your pull-request (that failed) to see a fresh report since the last report may be for someone else that did a later pull request, or for mainline code. If you add new code to fix your issue/problem, then take note that you need to check the next pull request in the Travis system. Travis issue numbers are different from GitHub issue numbers.

### CODE REVIEW

**Comment by skef** [comment]
I rebased this.

File: contrib/fonttools/showttf.c [hunk 1]
```diff
@@ -2484,10 +2484,13 @@ static void gposPairSubTable(FILE *ttf, int which, int stoffset, struct ttfinfo
...
     } else if ( subformat==2 ) {
 	printf( "\t   Class based kerning (not displayed)\n" );
-	PrintGlyphs( showCoverageTable(ttf,stoffset+coverage,-1),info);
+	glyphs = showCoverageTable(ttf,stoffset+coverage,-1);
+	PrintGlyphs(glyphs,info);
     } else {
 	printf( "\t   !!! unknown sub-table format !!!!\n" );
     }
+    if (glyphs)
+        free(glyphs);
```
Comment (skef) [inline contrib/fonttools/showttf.c]: `if (glyphs)` implies that `glyphs` could be NULL, which makes it appropriate to do that at the top.

Comment (skef) [reply contrib/fonttools/showttf.c]: Of course, if it *can't* be NULL you can just remove the check.

File: contrib/fonttools/showttf.c [hunk 2]
```diff
@@ -2595,7 +2598,8 @@ static void gsubSingleSubTable(FILE *ttf, int which, int stoffset, struct ttfinf
 		    val >= info->glyph_cnt ? "!!! Bad glyph !!!" : info->glyph_names == NULL ? "" : info->glyph_names[val]);
 	}
     }
-    free(glyphs);
+    if (glyphs)
+        free(glyphs);
```
Comment (skef) [inline contrib/fonttools/showttf.c]: Same here, either way.

File: contrib/fonttools/showttf.c [hunk 3]
```diff
@@ -2622,7 +2626,8 @@ static void gsubMultipleSubTable(FILE *ttf, int which, int stoffset, struct ttfi
 	}
 	putchar('\n');
     }
-    free(glyphs);
+    if (glyphs)
+        free(glyphs);
```
Comment (skef) [inline contrib/fonttools/showttf.c]: etc

File: contrib/fonttools/showttf.c [hunk 4]
```diff
@@ -2650,6 +2655,8 @@ static void gsubAlternateSubTable(FILE *ttf, int which, int stoffset, struct ttf
 	}
 	putchar('\n');
     }
+    if (glyphs)
+        free(glyphs);
```
Comment (skef) [inline contrib/fonttools/showttf.c]: etc

File: contrib/fonttools/showttf.c [hunk 5]
```diff
@@ -5849,6 +5860,8 @@ static int readcff(FILE *ttf,FILE *util, struct ttfinfo *info) {
 	    }
 	}
     }
+    if (fontnames)
+        free(fontnames);
```
Comment (skef) [inline contrib/fonttools/showttf.c]: etc

**Comment by skef** [comment]
You'll need to pull to get the effects of the rebase locally, either into a new branch or over the current one.

File: contrib/fonttools/showttf.c (hunk 5 above)
Comment (skef) [reply contrib/fonttools/showttf.c]: actually this one was OK.

**Comment by skef** [comment]
Alright, my suggested changes were not as advisable as I thought, but I went in and did a little work and no harm done.

**Comment by skef** [comment]
@jtanx Linux-full bombed out on this with:
```
$ if [ ! -z "$LINUX_FULL" ]; then make distcheck; fi
make  dist-gzip am__post_remove_distdir='@:'
make[1]: Entering directory '/home/travis/build/fontforge/fontforge'
make[1]: *** No rule to make target 'doc/README-Mac.html', needed by 'distdir'.  Stop.
make[1]: Leaving directory '/home/travis/build/fontforge/fontforge'
Makefile:2117: recipe for target 'dist' failed
make: *** [dist] Error 2
The command "if [ ! -z "$LINUX_FULL" ]; then make distcheck; fi" exited with 2.
```

**Comment by jtanx** [comment]
@skef yeah it's a side-effect of #3534, the old readmes need to be removed from a makefile.am somewhere

**Comment by jtanx** [comment]
#3546

**Comment by serval2412** [comment]
Thank you for all your feedback.
If I add a test before freeing it's because vars quoted by Coverity are initialized from functions which don't always malloc (eg showCoverageTable).

**Comment by skef** [comment]
a `malloc()`-paired `free()` is always NULL-tolerant in my experience.

**Comment by serval2412** [comment]
@skef: argh, you're right (see https://stackoverflow.com/questions/6084218/is-it-good-practice-to-free-a-null-pointer-in-c). I should have checked before adding these tests.

**Pull request merged**